zilencer/views.py

```diff
@@ -162,9 +162,11 @@ def register_remote_push_device(
     user_uuid: Optional[str] = REQ(default=None),
     token: str = REQ(),
     token_kind: int = REQ(json_validator=check_int),
-    ios_app_id: Optional[str] = None,
+    ios_app_id: Optional[str] = REQ(default=None),
```

Comment:
This doesn't do any sort of validation.  We should probably limit the acceptable options for this somehow before we start feeding arbitrary values into the database?

Reply:
Hmm interesting question.

The thing we'll do with this value is (as you see in a later commit) to set it as the APNs topic for notifications we send to this device. What that means is setting the HTTP header `apns-topic`, whose meaning is documented here:
https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns#4294479

I believe if this value is garbage, the only consequence that will happen is that APNs rejects that notification attempt — much the same as if the client had given us a random token it made up. The APNs error responses are documented here:
https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns
`TopicDisallowed` is the one I saw when first testing a zulip-flutter with iOS notification support, when my server was still always sending topic "org.zulip.Zulip". If the topic was instead garbage that didn't look like a plausible topic, the response might instead be `BadTopic`.

For caution, though, just in case something in the pipeline can be made confused or grumpy by a clever value here, we could limit it to something that looks plausibly like an app ID. I think /[a-zA-Z0-9.-]+/ would cover it.

Reply:
Couldn't we just restrict it to be one of the two official app IDs? I don't see a reason the bouncer would need to support any other values.

Reply:
Yeah, I was assuming we'd just hardcode a set of accepted values.

Reply:
`check_string_in` is the validator method for that use case.

Reply:
Hmm, thought I'd replied to this but apparently didn't post it.

We could do that, but it'd mean an additional thing to configure. It's another server setting, and then any time we want to have a different app ID — even just for an experiment — it requires going and propagating a setting change onto the push bouncer. And it's unnecessary, because there's no way we're going to send to an app that's not ours anyway: Apple already enforces that. So I'd prefer not to duplicate into our server settings a copy of the list of our app IDs that lives in our Apple developer account.